contracts/test/v0.7/KeeperRegistry.test.ts

```diff
@@ -19,7 +21,7 @@ const CHECK_GAS_OVERHEAD = BigNumber.from(170000);
 // Smart contract factories
 let linkTokenFactory: ContractFactory;
 let mockV3AggregatorFactory: ContractFactory;
-let keeperRegistryFactory: ContractFactory;
+let keeperRegistryFactory: KeeperRegistryFactory;
```

Comment:
Nice, strongly typed is definitely preferred.

I have found that typechain doesn't support a few things and have had to resort to `ContractFactory` in some cases. For example, it struggles with `try-catch` and raw `call`s sometimes, like [here in our internal repo](https://github.com/smartcontractkit/chainlink-ccip/pull/40/files#diff-692c36d592c0d30a9b71fcf2cce571ba6f98e44d6ec32579e4b8c55b02e4a0a5R27).

Be wary of this, especially when you get around to converting the `ValidatorProxy` tests.

Reply:
sounds good! I figured we can just convert them on an as-needed basis. And if we have any issues we can always fall back to the generic wrapper.